guides/advanced_guides/faceted_search.md

```diff
@@ -24,14 +24,15 @@ Faceting and filtering aim at being complementary;  facets narrows down the set
 
 ## Setting Up Facets
 
-Attributes that could be used as facets **must be declared at indexing time**.
+The first step in using facets is to chose which of your document <clientGlossary word="field" label="fields"/> will be used as facets. Fields with common values are the most suited for faceting (e.g., `genre`, `color`, `size` ).
 
-You can set up facets **through the API** via the [global settings route](/references/settings.md#update-settings).
-Attributes for which to enable faceting are defined in the `attributesForFaceting` list.
+For these fields to be used as facets during search, their <clientGlossary word="attribute" label="attributes"/> **have to be previously added to the settings**. In the settings, the chosen attributes must be added to the [`attributesForFaceting` list](/guides/advanced_guides/settings.md#attributes-for-faceting).
+It is a required step because facet needs to be properly processed and prepared by the engine to be usable. This process takes as much time as indexing all your documents.
 
-Any POST request on the `settings` route with the `attributesForFaceting` parameter set will overwrite the current value for `attributesForFaceting`: passing an empty array will remove all defined faceted attributes.
+The facet fields' value should be of type `string` or array of `string`. In some documents, if the value of the facet is `null`, it will be handled as if the document did not have that field.
```

Comment:
```suggestion
You can perform faceting on attributes that are either `String` or `[String]`, and null values are ignored.
```

Reply:
Would it be necessary to write null with quotes?